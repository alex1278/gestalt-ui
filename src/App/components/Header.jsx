import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const HeaderStyle = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  height: 56px;
  z-index: 19;
  background-image: radial-gradient(circle, ${props => props.theme.colors['$russian-black-50']} 0%, ${props => props.theme.colors['$russian-black-100']} 100%);
  color: white;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.1), 0 1px 1px 0 rgba(0, 0, 0, 0.1), 0 0 1px -4px rgba(0, 0, 0, 0.2);
`;

const LeftContent = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  height: 100%;

  button,
  i * {
    color: white;
  }

  #orgs-nav {
    height: 100%;
  }
`;

const Logo = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 69px;
  height: 100%;
`;

const RightContent = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding-right: 16px;

  button,
  i * {
    color: white;
  }
`;

class Header extends Component {
  render() {
    const { logo, logoVisible, leftContent, rightContent } = this.props;

    return (
      <HeaderStyle>
        {logo && logoVisible &&
        <Logo>
          {logo}
        </Logo>}
        <LeftContent>
          {leftContent}
        </LeftContent>

        <RightContent>
          {rightContent}
        </RightContent>
      </HeaderStyle>
    );
  }
}

Header.propTypes = {
  leftContent: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  rightContent: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  logo: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  logoVisible: PropTypes.bool,
};

Header.defaultProps = {
  leftContent: null,
  rightContent: null,
  logo: null,
  logoVisible: true,
};

export default Header;
