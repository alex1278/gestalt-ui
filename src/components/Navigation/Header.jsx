import styled, { withTheme } from 'styled-components';
import { Toolbar } from 'react-md';

const Header = styled(Toolbar)`
  z-index: 9999;
  background-color: ${props => props.theme.appHeaderColor};
`;

export default withTheme(Header);
