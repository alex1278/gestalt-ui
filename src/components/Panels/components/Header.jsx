import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexybox';
import styled, { css } from 'styled-components';
import ExpanderIcon from './ExpanderIcon';

const noShadowCSS = css`
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
`;

const TitleSection = styled(Col)`
  ${props => !props.noShadow && noShadowCSS};
  height: 100%;
  display: flex;
  align-items: center;
`;

const Title = styled.div`
  display: inline-block;
  font-size: 16px;
  color: ${props => props.theme.colors.fontTitle};
`;

const Icon = styled.div`
  display: inline-block;
  margin-right: 5px;
`;

const CountSection = styled(Col)`
  text-align: right;
  padding-right: 8px;
`;

const Bubble = styled.div`
  display: inline-block;
  min-width: 32px;
  height: 32px;
  padding: 8px;
  border-radius: 5px;
  text-align: center;
  background-color: ${props => props.theme.colors.primary.default};
  color: white;
`;

const HeaderStyle = styled.header`
  ${props => !props.expandable && !props.noShadow && 'padding-left: 8px'};
  height: 48px;
  font-weight: bold;
  user-select: none;

  &:hover {
    cursor: ${props => (!props.expandable ? 'inherit' : 'pointer')};
  }
`;

class Header extends PureComponent {
  static propTypes = {
    expandable: PropTypes.bool,
    expanded: PropTypes.bool,
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    count: PropTypes.oneOfType([PropTypes.number, PropTypes.bool]),
    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool]),
    noShadow: PropTypes.bool,
  };

  static defaultProps = {
    expandable: true,
    expanded: true,
    onClick: null,
    count: false,
    icon: false,
    noShadow: false,
  };

  render() {
    const { title, count, expandable, expanded, onClick, icon, noShadow } = this.props;
    const showCount = (count || count !== 0) && !expanded;

    return (
      <HeaderStyle expandable={expandable} onClick={onClick} noShadow={noShadow}>
        <Row gutter={1} center fill wrap="nowrap">
          <TitleSection flex={showCount ? 10 : 12}>
            {expandable && <ExpanderIcon isExpanded={expanded}>expand_more</ExpanderIcon>}
            {icon && <Icon>{icon}</Icon>}
            <Title small>{title}</Title>
          </TitleSection>
          {showCount &&
          <CountSection flex={2}>
            <Bubble>{count}</Bubble>
          </CountSection>}
        </Row>
      </HeaderStyle>
    );
  }
}

export default Header;
