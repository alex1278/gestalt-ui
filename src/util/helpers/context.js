/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
