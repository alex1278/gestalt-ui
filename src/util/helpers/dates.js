import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
