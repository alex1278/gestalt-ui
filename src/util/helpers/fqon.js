// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
