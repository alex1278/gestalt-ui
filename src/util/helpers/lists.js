export function insertItem(array, payload) {
  return [...array, payload];
}

export function removeItem(array, payload) {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a === payload), 1);

  return newArray;
}

export function removeItemById(array, id, keyField = 'id') {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a[keyField] === id), 1);

  return newArray;
}

export function updateItem(array, payload) {
  return array.map((item) => {
    if (item.id === payload.id) {
      return {
        ...payload,
      };
    }

    // Leave every other item unchanged
    return item;
  });
}
