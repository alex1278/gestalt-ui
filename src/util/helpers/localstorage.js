/**
 * getItem
 * @param {*} key
 */
export const getItem = (key) => {
  try {
    return localStorage.getItem(key);
  } catch (err) {
    return undefined;
  }
};

/**
 * saveItem
 * @param {*} key
 * @param {*} value
 */
export const saveItem = (key, value) => {
  try {
    localStorage.setItem(key, value);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};

/**
 * removeItem
 * @param {*} key
 * @param {*} value
 */
export const removeItem = (key) => {
  try {
    localStorage.removeItem(key);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};
