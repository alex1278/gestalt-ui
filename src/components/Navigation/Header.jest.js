import React from 'react';
import Header from './Header';
import { shallowWithTheme } from '../../../test/helpers';

describe('(Navigation) Header', () => {
  it('renders component without exploding', () => {
    const wrapper = shallowWithTheme(<Header />);

    expect(wrapper).toHaveLength(1);
  });

  it('renders a Header', () => {
    const wrapper = shallowWithTheme(<Header />);

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
