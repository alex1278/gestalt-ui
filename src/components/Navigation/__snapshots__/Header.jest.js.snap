// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Navigation) Header renders a Header 1`] = `
.c0 {
  z-index: 19;
  background-color: #1C1F2F;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 1px 5px 0 rgba(0,0,0,0.12),0 3px 1px -2px rgba(0,0,0,0.2);
}

<Paper
  className="md-toolbar c0"
  component="header"
  zDepth={0}
>
  <ToolbarTitle
    key="title"
  />
</Paper>
`;
